service/history/queues/executable.go

```diff
@@ -303,14 +317,20 @@ func (e *executableImpl) Ack() {
 	e.state = ctasks.TaskStateAcked
 
 	if e.shouldProcess {
+		e.metricsProvider.Timer(TaskLoadLatency).Record(
+			e.loadTime.Sub(e.GetVisibilityTime()),
+			metrics.QueueReaderIDTag(e.readerID),
```

Comment:
It looks like the readerID is only needed for metrics tag. Would it be possible to tag the metricsHandler with the readerID before creating the executable so executable don't need to know about the readerID?

Reply:
I am thinking that readerID is only relevant for metrics that including db latency (`task_latency_queue{,_nouserlatency}` and `task_latency_load`). It's weird that metrics like `task_latency_processing` also contains the readerID tag, as it has nothing to do with task loading.  It also adds unnecessary cardinality for those metrics.